lib/trento.pm

```diff
@@ -826,6 +827,52 @@ sub cluster_wait_status {
     die "Timeout waiting for the change" if !$done;
 }
 
+=head3 cluster_wait_status_by_regex
+
+Remotly run 'SAPHanaSR-showAttr' in a loop on $host, wait output that match using regular expression
+=over 3
+
+=item B<HOST> - Ansible name or filter for the remote host where to run 'SAPHanaSR-showAttr'
+
+=item B<TIMEOUT> - Max time to retry. Die if timeout
+
+=item B<REGULAR_EXPRESSION> - Regular expression to match the text to find
+
+=back
+=cut
+
+sub cluster_wait_status_by_regex {
+    my ($host, $timeout, $regular_expression) = @_;
+    $timeout //= bmwqemu::scale_timeout(300);
+    my $_monitor_start_time = time();
+    my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+    my $done;
+    croak "Not regular expression provided" if ($regular_expression eq "") ;
+    while ((time() - $_monitor_start_time <= $timeout) && (!$done)) {
+        sleep 30;        
+        my $show_attr = qesap_ansible_script_output(
+            cmd => 'SAPHanaSR-showAttr',
+            provider => $prov,
+            host => $host,
+            root => 1);
+
+        my @status = ();
+        for my $line (split("\n", $show_attr)) {
+            push @status , $line if ($line =~ $regular_expression);
+        }
+        if (scalar @status > 0) {
+            $done = 1;
+        }
+        else {
+            $done = 0;
+        }
```

Comment:
While code looks better with the `@status` list instead of the `%status` hash, I still don't see the need for it.

Basically `$done` will be set to one if there are elements in the list or to zero if the list is empty. List is declared empty and populated whenever the loop finds a line which matches `$regular_expression`. Then, why not simply do:
```
        for my $line (split("\n", $show_attr)) {
            $done = 1 if ($line =~ $regular_expression);
        }
```
And remove `@status` completely?

Reply:
good tip thx